src/pubsub/ua_pubsub_writer.c

```diff
@@ -2143,12 +2182,32 @@ generateNetworkMessage(UA_PubSubConnection *connection, UA_WriterGroup *wg,
 }
 
 static UA_StatusCode
-sendBufferedNetworkMessage(UA_Server *server, UA_PubSubConnection *connection,
+sendBufferedNetworkMessage(UA_Server *server, UA_WriterGroup *writerGroup,
```

Comment:
The function is called `sendBufferedNetworkMessage`. It should do exactly that and not more.

Please pull the encryption out of here and do it in the caller.
Likewise pull `UA_NetworkMessage_updateBufferedMessage` out of here.